SuspectBlue ip

SuspectBlue opened the pull request:
# Pelops II 

> "Hello from Pelops II. This computer has little space for me. Let me organize the data inside." - Pelops II

Welcome to Pelops II, your task management companion! 
This application helps you keep track of your _to-dos, deadlines, and events_ with ease.

## Features

Here's a quick rundown of what Pelops II can do:

* Add **To-Do** tasks
* Create tasks with specific **Deadlines** 
* Schedule **Events** with start and end times.
* Mark tasks as done or not done ✅
* Delete unwanted tasks
* Search for tasks

## How to Use

Follow these steps to get started with Pelops II:

1.  Download the latest release from [source](https://github.com/SuspectBlue/ip).
2.  Run the application from your terminal.
3.  Enter commands like: `todo <description>`, `deadline <description> /by <date/time>`, `event <description> /from <start_time> /to <end_time>`, `list`, `mark <index>`, `unmark <index>` and `delete <index>` to manage your tasks.

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
``` java
public class Main {
    public static void main(String[] args) throws IOException {
        new PelopsII("data").run();
    }
}
```
## Current Progress

- [X] Tasklist implementation
- [X] Storage Capabilities
- [ ] UI/UX Design

### Notes:
This task management companion is inspired by [Pelops II](https://wikizilla.org/wiki/Pelops_II) from [Godzilla Singular Point](https://godzilla.fandom.com/wiki/Godzilla_Singular_Point).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SuspectBlue committed:
Rename, Greet, Exit

SuspectBlue committed:
Echo

SuspectBlue committed:
Fixed bug with echo

SuspectBlue committed:
Add, List

SuspectBlue committed:
Mark as Done

SuspectBlue committed:
ToDos, Events, Deadlines

SuspectBlue committed:
Automated Text UI Testing

SuspectBlue committed:
Handle Errors

SuspectBlue committed:
Implemented Delete for TaskList

SuspectBlue committed:
add save feature for task list

SuspectBlue committed:
testing branch

SuspectBlue committed:
Merge branch 'branch-Level-7'
combine main with save feature

SuspectBlue committed:
Deadline.java: Update date format

SuspectBlue committed:
Merge branch 'branch-Level-8'
merge deadline date feature with master

SuspectBlue committed:
Add ui, storage, parse, tasklist, commands

SuspectBlue committed:
Organise files into packages

SuspectBlue committed:
Merge remote-tracking branch 'origin/add-gradle-support'
Merge gradle support for ip week 3

SuspectBlue committed:
Update gradle main class location

SuspectBlue committed:
Add tests to gradle

SuspectBlue committed:
Add Jar file to run

SuspectBlue committed:
Add JavaDocs for various classes

SuspectBlue committed:
Tweak code to match coding standard

SuspectBlue committed:
Add find feature in tasklist

SuspectBlue committed:
Merge branch 'branch-A-JavaDoc'
Add JavaDocs to various classes and functions

SuspectBlue committed:
Merge branch 'branch-A-CodingStandard'

SuspectBlue committed:
Merge branch 'branch-Level-9'

SuspectBlue committed:
Hide test messages and uncomment tests

TanXingZhong commented on src/main/java/pelopsii/exception/PelopsIIException.java:
Should this be extracted out and break it into more types of exceptions?  🤔

TanXingZhong commented on src/main/java/pelopsii/task/TaskList.java:
```diff
@@ -0,0 +1,177 @@
+     * @throws PelopsIIException If there is an error accessing or processing the task list.
+     */
+    public String find(String keyword) throws PelopsIIException {
+        StringBuilder sb = new StringBuilder();
```
I suggest renaming the variable such that it will be easier to understand. 🤔

TanXingZhong commented on src/main/java/pelopsii/command/HelpCommand.java:
```diff
@@ -0,0 +1,22 @@
+
+    @Override
+    public void execute() throws PelopsIIException {
+        StringBuilder sb = new StringBuilder();
```
Perhaps a more easier way to do this is just to create a string?🤔

TanXingZhong reviewed (commented):
LGTM! Just a few minor nits 😄

SuspectBlue committed:
Add Gui

SuspectBlue committed:
Merge branch 'branch-Level-10'
Merge master branch with level 10 GUI

SuspectBlue committed:
Update README.md

LsmnBmnc commented on src/main/java/pelopsii/parser/Parser.java:
```diff
@@ -0,0 +1,64 @@
+     * @return A Command object corresponding to the user input.
+     * @throws PelopsIIException If the input cannot be parsed into a valid command.
+     */
+    public static Command parse(String input) throws PelopsIIException {
```
I like your coding standard.

SuspectBlue committed:
Replace DaDuke with Pelops II

sunpterodactyl commented on src/main/java/pelopsii/command/EventCommand.java:
```diff
@@ -0,0 +1,40 @@
+import pelopsii.exception.PelopsIIException;
+import pelopsii.task.Event;
+
+public class EventCommand extends Command{
```
Might be fruitful to add in a short javadocs header here? I noticed the same issue in a few other classes.

sunpterodactyl commented on src/main/java/pelopsii/PelopsII.java:
```diff
@@ -0,0 +1,81 @@
+     *
+     * @throws IOException If an I/O error occurs during command processing.
+     */
+    public void run() throws IOException{
```
Maybe add in a space before the opening bracket?

sunpterodactyl commented on src/main/java/pelopsii/parser/Parser.java:
```diff
@@ -0,0 +1,64 @@
+    public static Command parse(String input) throws PelopsIIException {
+        String[] action = input.split(" ");
+        String command = action[0].toLowerCase();
+        switch (command) {
```
Nicely formatted!

sunpterodactyl reviewed (commented):
Very well done! Only nitpicking from me

SuspectBlue committed:
Add date to event and bugfix for incorrect inputs

SuspectBlue committed:
Add assertions and new jar file

Assertions added to ensure assumptions of the program holds true during
runtime.

To deal with the case where there is a bug in the code which would break
an assumption about the program.

As a step towards bug free code, let's add assertions to ensure
instantiated variables are not null and deleted tasks are not null.

SuspectBlue committed:
Improve code quality

Code quality for some commands and variables is difficult to understand.

Improving code quality improves readability and decreases complexity of
the program.

Let's improve the append functions for StringBuilders and convert all
static and final variables to be uppercase with spaces between each
word.

SuspectBlue committed:
Merge pull request #1 from SuspectBlue/branch-A-Assertions

Add assertions and new jar file

SuspectBlue committed:
Merge branch 'master' into branch-A-CodeQuality

SuspectBlue committed:
Merge pull request #2 from SuspectBlue/branch-A-CodeQuality

Improve code quality

SuspectBlue committed:
Add undo feature to the tasklist

User is unable to undo an expensive mistake. For example deleting an
event.

Allowing the user to undo the previous command improves the user
experience.

Let's add a new undo command to follow OOP principles and
add a undoTracker to track previous state.

SuspectBlue committed:
Merge branch 'branch-undo'

SuspectBlue committed:
Create gradle.yml

SuspectBlue committed:
Fix date issue

SuspectBlue committed:
Update test cases

SuspectBlue committed:
Fix date formatting

SuspectBlue committed:
Add undo command to help text

SuspectBlue committed:
Remove dead code

SuspectBlue committed:
Add README.md to docs

SuspectBlue committed:
Update JavaDocs

SuspectBlue committed:
Remove binary files from repo

SuspectBlue committed:
Update README.md

SuspectBlue committed:
Update empty task list message

SuspectBlue committed:
Update README.md file and help command text

SuspectBlue committed:
Fix find command bug
